src/components/node-list/node-list-items.js

```diff
@@ -176,14 +223,16 @@ export const getFilteredNodeItems = createSelector(
 
 /**
  * Get formatted list of sections
+ * @param {boolean} flag value of modularpipeline flag
  * @return {array} List of sections
  */
-export const getSections = createSelector(() =>
-  Object.keys(sidebar).map((name) => ({
+export const getSections = (flag) => {
+  const sidebarObject = sidebar(flag);
+  return Object.keys(sidebarObject).map((name) => ({
     name,
-    types: Object.values(sidebar[name]),
-  }))
-);
+    types: Object.values(sidebarObject[name]),
+  }));
+};
```

Comment:
I had set this up as a simple function given that it does not technically read anything from the state it is merely a function to return an object, but after our discussion yesterday I appreciate the potential benefits of using a selector ( which, however, it does raises the question on which should technically be be arranged and what shouldn't in our codebase, given that we do set up functions for calculations all over the codebase, with many that still do rely on the state but still structured as functions). 

regardless i can set it up as a selector.

Also speaking of, I've always wanted to raise this question - I wonder the reason behind setting up the selectors for the sidebar ( `node-list-items`) under the components folder rather than under the `selectors` folder>

Reply:
> Also speaking of, I've always wanted to raise this question - I wonder the reason behind setting up the selectors for the sidebar ( node-list-items) under the components folder rather than under the selectors folder>

I think the main reason is that they're highly specific to that component and not intended to be reusable across the whole app, where as those that live in `src/selectors` are designed to be generic and reusable. 

There's definitely other ways to get the performance benefits of selectors in pure react these days, but I don't think these were around at the time this was implemented.